File: 게임/q056/monster.js
function Monster(name,hp,attack,exp){ 
    /* 속성, 프로퍼티 */
    this.name = name;  // 이름
    this.hp = hp;    // 현재체력
    this.maxhp = hp;    // 최대체력
    this.attack = attack; // 공격력
    this.exp = 100;       //경험치
    /* 멤버함수, 멤버메소드 */
    this.info = function(){
        dw("["+this.name+"("+this.hp + "/" + this.maxhp+")]");
    }
}

File: eclipse/Chap01/q011/src/q011/Monster.java
package q011;

public class Monster {
	String name;
	int currentHp;
	int maxHp;
	int atk;
	public Monster(String name, int currentHp, int maxHp, int atk) {
		this.name = name;
		this.currentHp = currentHp;
		this.maxHp = maxHp;
		this.atk = atk;
	}
	void info() {
		String s =
				"[<" + name + ">"
				+ "(" + currentHp + "/" + maxHp + ")"
				+ "{atk:"+atk+"}]";
		System.out.println(s);						
	}
	int getRandomAtk() {
		int randomAtk = (int)(Math.random()*atk+1);
		return randomAtk; 
	}	
	
	
}